src/server.c

```diff
@@ -6121,15 +6121,47 @@ sds genRedisInfoString(const char *section) {
 }
 
 void infoCommand(client *c) {
-    char *section = c->argc == 2 ? c->argv[1]->ptr : "default";
+    if (c->argc == 1) {
 
-    if (c->argc > 2) {
-        addReplyErrorObject(c,shared.syntaxerr);
+        sds info = genRedisInfoString("default");
+        addReplyVerbatim(c,info,sdslen(info),"txt");
+        sdsfree(info);
+        return;
+    }
+
+    int defsections = 0, allsections = 0;
+    // first time find all/default flag
+    for (int i = 1; i < c->argc; i++) {
+
+        defsections = !strcasecmp(c->argv[i]->ptr,"default");
+        allsections = !strcasecmp(c->argv[i]->ptr,"all");
+    }
+
+    if (defsections || allsections) {
+
+        sds info = allsections ? genRedisInfoString("all") : genRedisInfoString("default");
+        addReplyVerbatim(c,info,sdslen(info),"txt");
+        sdsfree(info);
         return;
     }
-    sds info = genRedisInfoString(section);
+
+    sds info = sdsempty();
+    int lastValid = 0; 
+    // second time parse specific section flag
+    for (int i = 1; i < c->argc; i++) {
+
+        if (lastValid) {
+            info = sdscat(info,"\r\n");
+        }
+        sds sectionInfo = genRedisInfoString(c->argv[i]->ptr);
+        info = sdscatlen(info,sectionInfo,sdslen(sectionInfo));
+        lastValid = sdslen(sectionInfo) > 0 ? 1 : 0;
+        sdsfree(sectionInfo); 
+    }
```

Comment:
if someone asks for `INFO memory memroy` he'll get it twice.
which i suppose is no big deal.